route rule: Support suppress-prefix-length

cathay4t opened the pull request:
Adding support of ip route rule `suppress-prefix-length` which equal to
`suppress_prefixlength` option in `ip route` command.

Example:

```
route-rules:
  config:
    - ip-from: 192.168.3.2/32
      suppress-prefix-length: 0
      route-table: 200
    - ip-from: 2001:db8:b::/64
      suppress-prefix-length: 1
      route-table: 200
```

Integration test cases included.

liangwen12year commented on rust/src/lib/nm/settings/route_rule.rs:
```diff
@@ -72,6 +72,18 @@ pub(crate) fn gen_nm_ip_rules(
+                        ErrorKind::NotSupportedError,
+                        format!(
+                            "Specified suppress-prefix-length {v} is \
+                            on supported by NetworkManager: {e}"
```
`not supported by NetworkManager` ?

cathay4t replied:
Fixed. Thanks!

liangwen12year commented on tests/integration/testlib/iprule.py:
```diff
@@ -35,6 +37,7 @@ def ip_rule_exist_in_os(
     logging.debug(f"Current ip rules in OS: {result[1]}")
     assert result[0] == 0
     current_rules = json.loads(result[1])
+    print("Current", current_rules)
```
Is this on purpose or for debugging purpose?

cathay4t replied:
Debug. Removed. Thanks!

cathay4t committed:
route rule: Support suppress-prefix-length

Adding support of ip route rule `suppress-prefix-length` which equal to
`suppress_prefixlength` option in `ip route` command.

Example:

```
route-rules:
  config:
    - ip-from: 192.168.3.2/32
      suppress-prefix-length: 0
      route-table: 200
    - ip-from: 2001:db8:b::/64
      suppress-prefix-length: 1
      route-table: 200
```

Integration test cases included.

Signed-off-by: Gris Ge <fge@redhat.com>
